src/media.rs

```diff
@@ -32,78 +29,78 @@ pub fn generate_thumbnail_and_get_size(image: &[u8]) -> (Vec<u8>, (u32, u32)) {
 }
 
 /// Download file from servers and decrypt it
-pub fn download_file(file_info: FileInfo, media_type: MediaType, callback: Box<Fn(Result<Vec<u8>>) + Send + Sync>) {
-    thread::spawn(move || {
-        let mut file_enc = Cursor::new(Vec::with_capacity(file_info.size));
-
-        callback(reqwest::get(&file_info.url)
-            .map_err(|e| Error::with_chain(e, "could not load file"))
-            .and_then(|mut response| {
-                let status = response.status();
-                if status.is_success() {
-                    response.copy_to(&mut file_enc)
-                        .map_err(|e| Error::with_chain(e, "could not load file"))
-                } else {
-                    bail_untyped!{"received http status code {}", status.as_u16()}
-                }
-            })
-            .and_then(|_| crypto::decrypt_media_message(&file_info.key, media_type, &file_enc.into_inner())));
-    });
+pub async fn download_file(file_info: FileInfo, media_type: MediaType) -> Result<Vec<u8>> {
+    let client = reqwest::Client::builder()
+        .user_agent("curl/7.69.1")
+        .build()?;
+    let response = client
+        .get(&file_info.url)
+        .send()
+        .await?;
+
+    if response.status() != reqwest::StatusCode::from_u16(200).unwrap() {
+        error!("{:?}", response);
```

Comment:
I'm unhappy about libraries generating logs at the `error` loglevel; in my book at least, that's supposed to be reserved for application software. Perhaps stuff the `response` into a new error type (instead of the `Untyped` error below) and return it to the caller, so they can do with it what they like?